cpp/include/cudf/detail/aggregation/device_aggregators.cuh

```diff
@@ -154,6 +155,41 @@ struct update_target_element<Source, aggregation::SUM> {
   }
 };
 
+template <typename Source>
+  requires(cuda::std::is_same_v<Source, int64_t>)
+struct update_target_element<Source, aggregation::SUM_WITH_OVERFLOW> {
+  __device__ void operator()(mutable_column_device_view target,
+                             size_type target_index,
+                             column_device_view source,
+                             size_type source_index) const noexcept
+  {
+    // For SUM_WITH_OVERFLOW, target is a struct with sum value at child(0) and overflow flag at
+    // child(1)
+    auto sum_column      = target.child(0);
+    auto overflow_column = target.child(1);
+
+    auto const source_value = source.element<Source>(source_index);
+    auto const old_sum =
+      cudf::detail::atomic_add(&sum_column.element<int64_t>(target_index), source_value);
+
+    // Early exit if overflow is already set to avoid unnecessary overflow checking
+    if (overflow_column.element<bool>(target_index)) { return; }
+
+    // Check for overflow before performing the addition to avoid UB
+    // For positive overflow: old_sum > 0, source_value > 0, and old_sum > max - source_value
+    // For negative overflow: old_sum < 0, source_value < 0, and old_sum < min - source_value
```

Comment:
Maybe add a comment here that this could probably be replaced by CCCL function call once https://github.com/NVIDIA/cccl/pull/3755 is merged.